File: frontend/src/types/learning.ts
export interface TopicMeta {
  id: string;
  title: string;
  description: string;
  learning_objective: string;
  sort_order: number;
  estimated_minutes: number;
  completed: boolean;
}

export interface LearningModule {
  id: string;
  title: string;
  description: string;
  icon: string;
  sort_order: number;
  topics: TopicMeta[];
  progress_percent: number;
}

export interface LearningModuleListResponse {
  modules: LearningModule[];
  overall_progress_percent: number;
}

export interface TopicContent {
  topic_id: string;
  title: string;
  module_id: string;
  module_title: string;
  markdown: string;
  completed: boolean;
  estimated_minutes: number;
}

export interface LearningProgressSummary {
  total_topics: number;
  completed_topics: number;
  overall_percent: number;
  modules: {
    module_id: string;
    module_title: string;
    completed: number;
    total: number;
    progress_percent: number;
  }[];
}

File: backend/app/schemas/learning.py
from typing import Optional
from pydantic import BaseModel


class TopicMeta(BaseModel):
    id: str
    title: str
    description: str
    learning_objective: str
    sort_order: int
    estimated_minutes: int
    completed: bool = False


class LearningModule(BaseModel):
    id: str
    title: str
    description: str
    icon: str
    sort_order: int
    topics: list[TopicMeta]
    progress_percent: int = 0


class LearningModuleListResponse(BaseModel):
    modules: list[LearningModule]
    overall_progress_percent: int = 0


class TopicContent(BaseModel):
    topic_id: str
    title: str
    module_id: str
    module_title: str
    markdown: str
    completed: bool = False
    estimated_minutes: int = 5


class TopicProgressUpdate(BaseModel):
    completed: bool


class ModuleProgress(BaseModel):
    module_id: str
    module_title: str
    completed: int
    total: int
    progress_percent: int


class LearningProgressSummary(BaseModel):
    total_topics: int
    completed_topics: int
    overall_percent: int
    modules: list[ModuleProgress]
